osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ClearAlarmRegisterCommandExecutor.java

```diff
@@ -69,34 +69,48 @@ public AccessResultCode execute(
       final MessageMetadata messageMetadata)
       throws ProtocolAdapterException {
 
-    final AttributeAddress alarmRegister1AttributeAddress =
-        this.dlmsObjectConfigService.getAttributeAddress(
-            device, DlmsObjectType.ALARM_REGISTER_1, null);
-
-    final AccessResultCode resultCodeAlarmRegister1 =
-        this.executeForAlarmRegister(conn, alarmRegister1AttributeAddress);
-
-    if (resultCodeAlarmRegister1 == null) {
-      throw new ProtocolAdapterException("Error occurred for clear alarm register 1.");
+    // Clear alarm 1
+    final Optional<AccessResultCode> optionalResultCodeAlarm1 =
+        this.clearAlarnRegister(conn, device, DlmsObjectType.ALARM_REGISTER_1);
+    if (!optionalResultCodeAlarm1.isPresent()) {
+      throw new ProtocolAdapterException("Unable to find alarm 1.");
     }
+    final AccessResultCode resultCodeAlarmRegister1 = optionalResultCodeAlarm1.get();
     if (resultCodeAlarmRegister1 != AccessResultCode.SUCCESS) {
       return resultCodeAlarmRegister1;
     }
 
-    final Optional<AttributeAddress> optAlarmRegister2AttributeAddress =
-        this.dlmsObjectConfigService.findAttributeAddress(
-            device, DlmsObjectType.ALARM_REGISTER_2, null);
-
-    if (!optAlarmRegister2AttributeAddress.isPresent()) {
+    // Clear alarm 2
+    final Optional<AccessResultCode> optionalResultCodeAlarm2 =
+        this.clearAlarnRegister(conn, device, DlmsObjectType.ALARM_REGISTER_2);
+    if (!optionalResultCodeAlarm2.isPresent()) {
       return resultCodeAlarmRegister1;
+    }
+    final AccessResultCode resultCodeAlarmRegister2 = optionalResultCodeAlarm2.get();
```

Comment:
If the resultCode for register2 is not success and the resultCode for register3 is success, then the resultCode returned by this function is success. I would expect it to return the failure result code.